packages/e2e-test-utils-playwright/src/editor/index.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import type { Browser, Page, BrowserContext, Frame } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { clickBlockOptionsMenuItem } from './click-block-options-menu-item';
+import { clickBlockToolbarButton } from './click-block-toolbar-button';
+import { getEditedPostContent } from './get-edited-post-content';
+import { insertBlock } from './insert-block';
+import { openDocumentSettingsSidebar } from './open-document-settings-sidebar';
+import { openPreviewPage } from './preview';
+import { selectBlockByClientId } from './select-block-by-client-id';
+import { showBlockToolbar } from './show-block-toolbar';
+import { saveSiteEditorEntities } from './site-editor';
+
+interface EditorConstructorParams {
+	page: Page;
+	hasIframe?: boolean;
+}
+
+export class Editor {
+	browser: Browser;
+	page: Page;
+	context: BrowserContext;
+	#hasIframe: boolean;
+
+	constructor( { page, hasIframe = false }: EditorConstructorParams ) {
+		this.page = page;
+		this.context = page.context();
+		this.browser = this.context.browser()!;
+		this.#hasIframe = hasIframe;
+	}
+
+	get canvas(): Frame | Page {
```

Comment:
I wonder if we could also return `PageUtils` or `Editor` here. So that in tests we could do something like:

```js
const { page /* or frame */, pageUtils, editor, ...etc } = editor.canvas;
```

We can then chain all the existing utils with an explicit API. I haven't looked into it enough to know if it's possible though 😅.

Reply:
Would you be able to add some more detail on that? Would the goal be to use pageUtils within the editor canvas?

Reply:
Yep, exactly. I think it's still a problem, right? Though it's not really related to this PR though and we can discuss it in another issue :).

Reply:
It's tricky because we'd have to pass the appropriate `frame` or `page` into `pageUtils`, possibly instantiating it on the fly within the `canvas` getter, and each util would have to be written in a way that takes that into account.

Looking at the existing puppeteer utils, there aren't many functions that would benefit from this. Possibly only `enableObserveFocusLoss` and `disableObserveFocusLoss` because I think they'd have to listen inside the iframe.

It may be easier to do something like `pageUtils.enableObserveFocusLoss( editor.canvas )`.